tool/teleport/common/teleport.go

```diff
@@ -237,13 +251,74 @@ func onStatus() error {
 	return nil
 }
 
+type dumpFlags struct {
+	config.SampleFlags
+	output string
+}
+
+func (flags *dumpFlags) CheckAndSetDefaults() error {
+	if flags.output == "" || flags.output == teleport.SchemeFile {
+		flags.output = teleport.SchemeFile + "://" + defaults.ConfigFilePath
+	} else if flags.output == teleport.SchemeStdout {
+		flags.output = teleport.SchemeStdout + "://"
+	}
+	return nil
+}
+
 // onConfigDump is the handler for "configure" CLI command
-func onConfigDump() error {
-	sfc, err := config.MakeSampleFileConfig()
+func onConfigDump(flags dumpFlags) error {
+	if err := flags.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+	uri, err := url.Parse(flags.output)
+	if err != nil {
+		return trace.BadParameter("could not parse output value %q, use --output=%q",
+			flags.output, defaults.ConfigFilePath)
+	}
```

Comment:
Minor: Pass through https://golang.org/pkg/path/filepath/#EvalSymlinks to resolve symlinks for the user.